pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -218,11 +218,13 @@ func (spc *realStatefulPodControl) createPersistentVolumeClaims(set *apps.Statef
 		_, err := spc.pvcLister.PersistentVolumeClaims(claim.Namespace).Get(claim.Name)
 		switch {
 		case apierrors.IsNotFound(err):
-			_, err := spc.client.Core().PersistentVolumeClaims(claim.Namespace).Create(&claim)
+			created, err := spc.createPersistentVolumeClaim(set, &claim)
 			if err != nil {
 				errs = append(errs, fmt.Errorf("Failed to create PVC %s: %s", claim.Name, err))
 			}
-			spc.recordClaimEvent("create", set, pod, &claim, err)
+			if created || err != nil {
```

Comment:
@kargakis the code in master right now always records an event after it attempts to create the PVC. This change only records if it succeeds, or if the error is something other than already-exists. Is that not what you want?